cmd/snap-confine/ns-support.c

```diff
@@ -319,6 +346,30 @@ static bool should_discard_current_ns(dev_t base_snap_dev)
 	return true;
 }
 
+// Inspect the namespace and check if we should discard it.
+static bool should_discard_current_ns(const struct sc_invocation *inv,
+                                      dev_t base_snap_dev)
+{
+	sc_mountinfo *mi SC_CLEANUP(sc_cleanup_mountinfo) = NULL;
+
+	mi = sc_parse_mountinfo(NULL);
+	if (mi == NULL) {
+		die("cannot parse mountinfo of the current process");
+	}
+
+	// The namespace may become "stale" when the rootfs is not the same
+	// device we found above. This will happen whenever the base snap is
+	// refreshed since the namespace was first created.
+	if (base_snap_device_changed(mi, base_snap_dev)) return true;
```

Comment:
(nitpick) maybe our code formatter did that, but I would prefer if this return would be in a new line (and below)